central/config/datastore/datastore.go

```diff
@@ -81,15 +143,21 @@ func (d *datastoreImpl) UpsertConfig(ctx context.Context, config *storage.Config
 		}
 	}
 
-	return d.store.Upsert(ctx, config)
+	upsertErr := d.store.Upsert(ctx, config)
```

Comment:
nitpick: It would be beneficial if we could unit test that `UpsertConfig` invalidates the cache.

Reply:
Changed `TestAllowsUpdate` to cover that.